File: python/cgi-bin/mastergame.py
import combinaison

class MasterGame:
    def __init__(self, _nbColors, _nbPositions):
        self.nbColors = _nbColors
        self.nbPositions = _nbPositions
        self.secretCombinaison = combinaison.Combinaison([0 for i in range(_nbPositions)])

    def setSecretCombinaison(self, combi):
        #print("setSecretCombinaison", combi.pColorTable)
        self.secretCombinaison = combi.clone()

    def getSecretCombinaison(self):
        return  self.secretCombinaison

    SecretCombinaison = property(getSecretCombinaison, setSecretCombinaison)

    def generateSecretCombinaison(self):
        self.secretCombinaison.random(m_nbColors)

    def getCorrection(self, referenceCombinaison, combiTotest):
        """returns (blackPigs,whitePigs)"""
        return referenceCombinaison.getCorrection(combiTotest);

    def getCorrectionOfSecretCombinaison(self, combiTotest):
        """returns (blackPigs,whitePigs)"""
        return self.secretCombinaison.getCorrection(combiTotest);

File: c/src/mastergame.cpp
//
// C++ Implementation: mastersolver
//
// Description: 
//
//
// Author:  <>, (C) 2007
//
// Copyright: See COPYING file that comes with this distribution
//
//
#include "mastersolver.h"
#include "combinaison.h"

namespace mastermind {

    MasterGame::MasterGame(tColor nbColors, U32 nbPositions) noexcept : m_nbColors(nbColors), m_nbPositions(nbPositions), m_secretCombinaison(nbPositions) {

    }

    void MasterGame::setSecretCombinaison(Combinaison &combi) noexcept {
        m_secretCombinaison = combi; //overloaded operator=
    }

    const Combinaison& MasterGame::getSecretCombinaison() const noexcept {
        return m_secretCombinaison;
    }

    void MasterGame::generateSecretCombinaison() noexcept {
        m_secretCombinaison.random(m_nbColors);
    }

    void MasterGame::getCorrection(const Combinaison &referenceCombinaison, const Combinaison &combiTotest, U32 &blackPigs, U32 &whitePigs) const noexcept {
        referenceCombinaison.getCorrection(combiTotest, blackPigs, whitePigs);
    }

    void MasterGame::getCorrectionOfSecretCombinaison(const Combinaison &combiTotest, U32 &blackPigs, U32 &whitePigs) const noexcept {
        m_secretCombinaison.getCorrection(combiTotest, blackPigs, whitePigs);
    }

}//end namespace
